payments-ui-core/src/test/java/com/stripe/android/ui/core/elements/CardNumberControllerTest.kt

```diff
@@ -434,6 +436,48 @@ internal class CardNumberControllerTest {
         verify(eventReporter, times(1)).onCardNumberCompleted()
     }
 
+    @Test
+    fun `on disallowed card brand entered, should report event`() = runTest {
+        val eventReporter: CardNumberCompletedEventReporter = mock()
```

Comment:
Ideally we could use fakes, instead of mocks.

Reply:
I'm happy to pair on this, but I think this test would be easier to read, and a better overall test if we used fakes with [turbines](https://github.com/cashapp/turbine).